pkg/dynamodb/dynamo_api.go

```diff
@@ -0,0 +1,419 @@
+package dynamodb
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"github.com/aws/aws-sdk-go-v2/feature/dynamodb/attributevalue"
+
+	"time"
+
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/feature/dynamodb/expression"
+	"github.com/aws/aws-sdk-go-v2/service/dynamodb"
+	"github.com/aws/aws-sdk-go-v2/service/dynamodb/types"
+)
+
+type Keys struct {
+	PartitionKey string `dynamodbav:"pk"`
+	SortKey      string `dynamodbav:"sk"`
+}
+
+type GetItemParams struct {
+	KeyCondition   Keys
+	ConsistentRead bool
+}
+
+type QueryItemsParams struct {
+	ProjectionKeys   []string
+	FilterParamsList []ItemParams
+	PartitionKey     string
+	SortKey          ItemParams
+	ConsistentRead   bool
+}
+
+type ItemParams struct {
+	KeyName       string
+	ConditionName string
+	Value         interface{}
+}
+
+type DeleteItemParams struct {
+	KeyCondition Keys
+}
+
+type BatchGetItemParams struct {
+	KeyCondition   []Keys
+	ProjectionKeys []string
+	ConsistentRead bool
+}
+
+func (dynamoClient *DynamoClient) GetItem(ctx context.Context, inputRequest GetItemParams, outputItem interface{}) (interface{}, error) {
+	row, err := marshalMap(inputRequest.KeyCondition)
+
+	if err != nil {
+		return nil, err
+	}
+
+	input := &dynamodb.GetItemInput{
+		TableName:      aws.String(dynamoClient.Table),
+		Key:            row,
+		ConsistentRead: &inputRequest.ConsistentRead,
+	}
+	result, err := dynamoClient.getItem(ctx, input)
+	if err != nil {
+		return nil, err
+	}
+
+	if outputItem == nil {
+		return result.Item, err
+	}
+	err = unmarshalMap(result.Item, outputItem)
+
+	return outputItem, err
+}
+
+func (dynamoClient *DynamoClient) PutItem(ctx context.Context, values interface{}) error {
+	row, err := marshalMap(values)
+
+	if err != nil {
+		return err
+	}
```

Comment:
**Suggestion:** In the `PutItem` method, consider checking if `values` is nil before calling `marshalMap` to avoid potential nil pointer dereference errors. [possible issue, importance: 9]
```suggestion
 if values == nil {
     return errors.New("values cannot be nil")
 }
 row, err := marshalMap(values)

 if err != nil {
     return err
 }
```